src/Voron/Data/BTrees/TreePageSplitter.cs

```diff
@@ -143,7 +144,7 @@ private void RecompressPageIfNeeded(bool wasModified)
             if (_pageDecompressed == null)
                 return;
             _pageDecompressed.CopyToOriginal(_tx, defragRequired: false, wasModified: wasModified, _tree);
-            _tree.DecompressionsCache.Invalidate(_pageDecompressed.PageNumber, DecompressionUsage.Read);
+            _tree.DecompressionsCache.Invalidate(_pageDecompressed.PageNumber, WriteDecompressionUsage);
```

Comment:
and here we need to invalidate it accordingly - also with with passing `DecompressionUsage.Write`